src/RequestExecutor/Request.spec.ts

```diff
@@ -0,0 +1,135 @@
+import 'reflect-metadata';
+import { VirtualScripts } from '../Scripts';
+import { RequestExecutorOptions } from './RequestExecutorOptions';
+import type { RequestExecutor } from './RequestExecutor';
+import type { Request } from './Request';
+import { WsRequestExecutor } from './WsRequestExecutor';
+import { instance, mock, reset, spy, when } from 'ts-mockito';
+
+type RequestType = 'http' | 'ws';
+
+describe('Request', () => {
```

Comment:
These tests below are more like integration tests. Spec tests should be isolated meaning do not depend on their clients(in this case HttpRequestExecutor and WsRequestExecutor) or any concrete implementations in general.
Test suit should be named the same as class/function name, each tested method should have its own nested describe block

For this task, `setCerts` method should be covered